lib/bolt/cli.rb

```diff
@@ -551,6 +557,114 @@ def list_groups
       outputter.print_groups(groups)
     end
 
+    def new_plan(plan_name)
+      @logger.warn("Command 'bolt plan new' is experimental and subject to breaking changes.")
+
+      if config.project.name.nil?
+        raise Bolt::Error.new(
+          "Project directory '#{config.project.path}' is not a named project. Unable to create "\
+          "a project-level plan. To name a project, set the 'name' key in a 'bolt-project.yaml' "\
+          "configuration file.",
+          "bolt/unnamed-project-error"
+        )
+      end
+
+      if plan_name !~ Bolt::Project::NAME_REGEX
+        message = <<~MESSAGE.chomp
+          Invalid plan name '#{plan_name}'. Plan names are composed of one or more name segments
+          separated by double colons '::'.
+          
+          Each name segment must begin with a lowercase letter, and may only include lowercase
+          letters, digits, and underscores.
+          
+          Examples of valid plan names:
+              - myproject
+              - my_project::my_plan
+
+          Examples of invalid plan names:
+              - MyProject
+              - my-project::my-plan
+        MESSAGE
+
+        raise Bolt::ValidationError, message
+      end
+
+      name_segments = plan_name.split('::')
+
+      unless name_segments.shift == config.project.name
+        message = "First segment of plan name '#{plan_name}' must match project name '#{config.project.name}'. "\
+                  "Did you mean '#{config.project.name}::#{plan_name}'?"
+
+        raise Bolt::ValidationError, message
+      end
+
+      # If the plan name is just the project name, then create an 'init' plan.
+      # Otherwise, use the last name segment for the plan's filename.
+      basename = name_segments.empty? ? 'init' : name_segments.pop
+      dir_path = config.project.plans_path.join(*name_segments)
+
+      %w[pp yaml].each do |ext|
+        next unless (path = config.project.plans_path + "#{basename}.#{ext}").exist?
+        raise Bolt::Error.new(
+          "Detected existing plan '#{plan_name}' at '#{path}'. Unable to create new plan.",
```

Comment:
The tone of this message is a bit technical compared to some of the others. Maybe just something like `A plan already exists at '#{path}', nothing to do`?